File: continuous_numbers.rb
def continuous_numbers(arr)
    return arr.length if (arr.length <= 1)
      
    continuous_count = 1
    a_pointer = 0
    b_pointer = 1
    
    while (b_pointer < arr.length)
      if (arr[a_pointer] === arr[b_pointer])
        b_pointer += 1
        continuous_count = [(b_pointer - a_pointer), continuous_count].max
      else
        a_pointer = b_pointer
        b_pointer += 1
      end
    end
    return continuous_count
end

File: continuousNumbers.js
function continuousNumbers(arr){
    if (arr.length <= 1) return arr.length
  
    let continuousCount = 1
    let aPointer = 0
    let bPointer = 1
    
    while (bPointer < arr.length){
      if (arr[aPointer] === arr[bPointer]){
        bPointer += 1
        continuousCount = Math.max((bPointer - aPointer), continuousCount)
      } else {
        aPointer = bPointer
        bPointer += 1
      }
    }
    return continuousCount
}

module.exports = continuousNumbers;